napari/utils/_injection/_accessors.py

```diff
@@ -0,0 +1,97 @@
+from typing import Callable, Dict, Optional, Type, TypeVar
+
+from ... import components, layers, viewer
+
+T = TypeVar("T")
+_NULL = object()
+
+
+def _access_viewer() -> Optional[viewer.Viewer]:
+    return get_accessor(viewer.Viewer)()
+
+
+def _get_active_layer() -> Optional[layers.Layer]:
+    return v.layers.selection.active if (v := _access_viewer()) else None
+
+
+def _get_active_layer_list() -> Optional[components.LayerList]:
+    return v.layers if (v := _access_viewer()) else None
+
+
+# registry of Type -> "accessor function"
+# where each value is a function that is capable
+# of retrieving an instance of its corresponding key type.
+_ACCESSORS: Dict[Type, Callable[..., Optional[object]]] = {
+    layers.Layer: _get_active_layer,
+    viewer.Viewer: viewer.current_viewer,
+    components.LayerList: _get_active_layer_list,
+}
+
+
+def get_accessor(type_: Type[T]) -> Optional[Callable[..., Optional[T]]]:
+    """Return object accessor function given a type.
+
+    An object accessor is a function that returns an instance of a
+    particular object type. For example, given type `napari.Viewer`, we return
+    a function that can be called to get the current viewer.
+
+    This is a form of dependency injection, and, along with
+    `inject_napari_dependencies`, allows us to inject current napari objects
+    into functions based on type hints.
+    """
+    if type_ in _ACCESSORS:
+        return _ACCESSORS[type_]
+
+    if isinstance(type_, type):
+        for key, val in _ACCESSORS.items():
+            if issubclass(type_, key):
+                return val  # type: ignore [return-type]
+    return None
+
+
+class set_accessor:
```

Comment:
super minor: slight preference for `set_accessors` given that the first argument is a dict

same goes for processors

Reply:
done